lib/apiservers/engine/backends/system.go

```diff
@@ -381,3 +413,46 @@ func FetchVolumeStores(client *client.PortLayer) (string, error) {
 
 	return strings.Join(res.Payload.Stores, " "), nil
 }
+
+// RegistrySetContains checks if a URL is within the set of insecure registries
+func RegistrySetContains(registrySet []url.URL, remoteURL *url.URL) bool {
+	containedOK := false
+	for _, rs := range registrySet {
+		// if CIDR was provided for insecure registry range, check if login URL within that ip net
+		if strings.Index(rs.Host, "/") != -1 {
+			remoteIP := net.ParseIP(remoteURL.Hostname())
```

Comment:
this line can be moved out the registrySet for loop, to avoid duplicate parsing in loop.